src/sentry/api/bases/organization_events.py

```diff
@@ -118,8 +118,14 @@ def get_snuba_dataclass(
                     "organizations:global-views", organization, actor=request.user
                 )
                 fetching_replay_data = request.headers.get("X-Sentry-Replay-Request") == "1"
-
-                if not has_global_views and len(params.projects) > 1 and not fetching_replay_data:
+                if (
+                    not has_global_views
+                    and len(params.projects) > 1
+                    and not fetching_replay_data
+                    # If a developer can view issues of a project they do not belong to
+                    # via opem membership, we will also allow the endpoint to return events for it
+                    and not organization.flags.allow_joinleave
+                ):
```

Comment:
nit: bit more readable?   

```suggestion
                if not any(
                    has_global_views,
                    len(params.projects) <= 1,
                    fetching_replay_data, 
                    # If a developer can view issues of a project they do not belong to
                    # via opem membership, we will also allow the endpoint to return events for it
                    organization.flags.allow_joinleave, 
                ):
```